WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -185,7 +185,7 @@ extension PostEditor where Self: UIViewController {
     func cancelEditing() {
         stopEditing()
 
-        if post.canSave() && post.hasUnsavedChanges() {
+        if post.canSave() && !post.isDraft() {
```

Comment:
removed `post.hasUnsavedChanges()` because we're already checking it in `canSave()`

Reply:
Removing the check for `hasUnsavedChanges` makes sense.  I'm not sure I understand why we'd want to check `!post.isDraft()`.  I think we'd want to prompt a user for unsaved changes before dismissing the editor. With this check any unsaved changes are just discard them outright.  What was the rationale for the change?  Was it trying to address some other issue?

Reply:
I was following @koke's comment saying:

> Existing the editor after preview (upload) would not show the unsaved changes action sheet.

I think this should be fine. If you look at Gutenberg or any of the older web editors, I think they start autosaving drafts when you type a bit and only ask you about unsaved changes if you changed anything since the last autosave. As long as the post is a draft, I think it’s OK to save things. IIRC WPAndroid is saving on closing the editor without even asking.

Since we are doing autosave it made sense to me. Otherwise the action sheet that is currently displayed has 2 options: continue editing or discard. If you think that we should show the action sheet I will update the code to show the option for "save and exit" that is no longer available because we are saving the draft.

Were the changes discarded for you? They seem to be saved for me.

Reply:
Thanks for the background!

Yeah, the changes were discarded for me.  I was testing very quickly so I might have just been too fast for autosave.  This suggests that a user who makes a series of edits to a draft and quickly dismisses the editor might loose some of their changes--anything they typed after the final autosave.   To be safe, maybe we'd either want to keep the prompt, or to force a final save for drafts.   Thoughts?

Reply:
In that case I will update.